src/Flux/neural_de.jl

```diff
@@ -1,22 +1,11 @@
-neural_ode_reduction(sol) = Array(sol)
-neural_ode(x,model,tspan,args...;kwargs...) = neural_ode(x,model,tspan,
-                                                         diffeq_adjoint,
-                                                         args...;kwargs...)
-function neural_ode(x,model,tspan,
-                    ad_func::Function,
+function neural_ode(model,x,tspan,
                     args...;kwargs...)
-  p = Flux.data(destructure(model))
+  Tracker.istracked(x) && error("u0 is not currently differentiable.")
+  p = destructure(model)
   dudt_(du,u::TrackedArray,p,t) = du .= restructure(model,p)(u)
   dudt_(du,u::AbstractArray,p,t) = du .= Flux.data(restructure(model,p)(u))
   prob = ODEProblem(dudt_,x,tspan,p)
-
-  if ad_func === diffeq_adjoint

```

Comment:
why remove the other options?

Reply:
The other options don't really make sense here. Forward mode is never going to be useful for a neural ODE and for reverse mode will work without the destructure/restructure shenanigans (which add unnecessary overhead). Making AD work with the same signature is probably worthwhile but we can follow up with that.